packages/x-components/src/x-modules/facets/components/facets/__tests__/facets-provider.spec.ts

```diff
@@ -129,6 +130,8 @@ function renderFacetsProviderComponent({
   stateFacets = {},
   providedFacets
 }: FacetsRenderOptions = {}): FacetsComponentAPI {
+  DefaultFacetsService.instance = new DefaultFacetsService(new FilterEntityFactory());
```

Comment:
Just to give some context here. What it's the difference of passing the `FilterEntityFactory`?

Reply:
Both the facets service and the entity factory are singletons, the problem is that they are stateless singletons, which is an issue to test different flow. In this line we are "resetting" the singleton by creating new instances.